README.md

```diff
@@ -2,15 +2,75 @@
 
 ## Intro
 
-This repository contains the backend logic for the dfusion exchange based on the specification, see [here](github.com/gnosis/dex-research)
+This repository contains the backend logic for the dfusion exchange based on [this specification](github.com/gnosis/dex-research).
 
-## Architecture
+It contains two sub-projects that both implement the market mechanism described above in different ways. An fully on-chain solution with instant finality but limited scalability (referred to as "StableX") and a preliminary version that intends to achieves scalability by offloading computation and data-storage off-chain using an [optimistic roll-up](https://medium.com/plasma-group/ethereum-smart-contracts-in-l2-optimistic-rollup-2c1cef2ec537) approach. The latter is in early development stage and not yet ready for use.
 
-<p align="center">
-  <img src="documentation/architecture.png" alt="dex-services architecture" width="500">
-</p>
+## Getting Started
+
+### Requirements
+
+- Rust (stable)
+- NodeJS <=11.0, starting with version 12 some deprecated APIs were removed that cause `scrypt`, `keccak`, `secp256k1`, and `sha3` packages to fail to build
+- Docker and Docker-compose (stable)
+
+The project may work with other versions of these tools but they are not tested.
+
+### Installation
+
+Clone the repository, its submodule, and run the container
+```bash
+git clone git@github.com:gnosis/dex-services.git
+cd dex-services
+git submodule update --init
+docker-compose up -d ganache-cli
+(cd dex-contracts && yarn && npx truffle compile && npm run networks-inject && npx truffle migrate)
```

Comment:
For me is fine, and if it adds too much complication we can forget about it. But is a bit strange we require docker, but for this parts we use something else.

I think docker allows to map the source code of the contracts to the container, and also to connect to the ganache launched in line 26.

Reply:
So you are suggesting to change the docker entry-point to run the truffle commands as a first step in the container before running the rust binary. That would work and could save the extra command.

However, generally it shouldn't be the responsibility of the service to compile or migrate any contracts. It's normal mode of operation is expecting that those have already been deployed (which happens here outside of the container). It would also require the node to have node installed.
Maybe we can add this for the ganache build and not use it for rinkeby/mainnet. I will take a look and change the README if that procedure changes.

Reply:
I think this should be changed to `(cd dex-contracts && yarn && yarn prepack && npx truffle migrate)`. That way if the build process slightly changes, we won't have to remember to update the README.

Reply:
In reality I was thinking more of running a `docker-compose run`

Anyways, it's not important for now, at least for me. Let's see the feedback we get from YK and others